### ISSUE TITLE
Rabbitmqadmin doesn't use management_ip_address parameter

### ISSUE THREAD

**Issue by geekix** [voxpupuli/puppet-rabbitmq#729]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.8
- Ruby: 2.1.9
- Distribution: Debian 8
- Module version: 8.3.0

## How to reproduce (e.g Puppet code you use)

In Hiera : rabbitmq::management_ip_address: '1.2.3.4'

## What are you seeing

When the management_ip_address parameter is set (and different then 127.0.0.1), I get a "connection refused" error message on rabbitmqadmin run. Everything is fine when management_ip_address is not set or set to "127.0.0.1"

It looks like rabbitmqadmin doesn't use the management_ip_address parameter to contact rabbitmq and always assume that rabbitmq responds on the localhost address.

Did I miss something ?

## What behaviour did you expect instead

Rabbitmqadmin should use the management ip address when the parameter is explicitly set.

## Output log

Connection refused for rabbitmqadmin :

`/usr/local/bin/rabbitmqadmin declare exchange --vhost=xxxxxxx --user=xxxxxxx --password=xxxxxxx name=xxxxxxx type=direct internal=false durable=true auto_delete=false arguments={} -c /etc/rabbitmq/rabbitmqadmin.conf
Traceback (most recent call last):
  File "/usr/local/bin/rabbitmqadmin", line 523, in http
    conn.request(method, path, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1039, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1073, in _send_request
    self.endheaders(body)
  File "/usr/lib/python2.7/httplib.py", line 1035, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 879, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 841, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 822, in connect
    self.timeout, self.source_address)
  File "/usr/lib/python2.7/socket.py", line 571, in create_connection
    raise err
error: [Errno 111] Connection refused
*** Could not connect: [Errno 111] Connection refused`

Thanks for your help

**Comment by wyardley** [comment]
I'm not 100% sure, you'd think it could get the host from the config file, but maybe `-H [hostname]` needs to be passed? This isn't something I'll be able to look at / fix, but I'd be happy to review if someone can come up with a PR for it. It could end up being pretty difficult, since those arguments are set separately in each of the providers.

**Comment by geekix** [comment]
Hello,

Thanks for your quick answer. After looking a bit more in the module code, I found that rabbitmqadmin is not using the management_ip_address parameter but management_hostname.

I did set the management_hostname parameter to the same value as management_ip_address and I don't have the issue anymore.

so if I understand correctly, management_ip_address manages the listener and management_hostname manages rabbitmqadmin "-H" option.

Thanks for your help =)
